packages/cells/src/model.ts

```diff
@@ -295,25 +295,55 @@ export class CellModel extends CodeEditor.Model implements ICellModel {
     event: IObservableJSON.IChangedArgs
   ): void {
     const metadata = this.nbcell.getMetadata();
-    this._mutex(() => {
+    this._modeDBMutex(() => {
       switch (event.type) {
         case 'add':
-          metadata[event.key] = event.newValue as any;
+          this._changeCellMetata(metadata, event);
+          break;
+        case 'change':
+          this._changeCellMetata(metadata, event);
           break;
         case 'remove':
           delete metadata[event.key];
           break;
-        case 'change':
-          metadata[event.key] = event.newValue as any;
-          break;
         default:
-          metadata[event.key] = event.newValue as any;
-          break;
+          throw new Error(`Invalid event type: ${event.type}`);
       }
       this.nbcell.setMetadata(metadata);
     });
   }
 
+  private _changeCellMetata(
+    metadata: Partial<nbmodel.ISharedBaseCellMetada>,
+    event: IObservableJSON.IChangedArgs
+  ) {
+    switch (event.key) {
+      case 'jupyter':
+        metadata.jupyter = event.newValue as any;
+        break;
+      case 'collapsed':
+        metadata.collapsed = event.newValue as any;
+        break;
+      case 'name':
+        metadata.name = event.newValue as any;
+        break;
+      case 'scrolled':
+        metadata.scrolled = event.newValue as any;
+        break;
+      case 'tags':
+        metadata.tags = event.newValue as any;
+        break;
+      case 'trusted':
+        metadata.trusted = event.newValue as any;
+        break;
+      case 'slideshow':
+        metadata.slideshow = event.newValue as any;
+        break;
+      default:
+        throw new Error(`Invalid event key: ${event.key}`);
```

Comment:
I think we can simply ignore this case. There are at least a few custom modeldb fields that could be set here. Maybe extensions add custom properties as well. So we don't want to throw an error (and possibly break synchronization) when an unknown key is added.

Reply:
or should we support non standard metadata that can be set via the UI?